lib/index.js

```diff
@@ -80,25 +90,17 @@ var loadPolicies = function(server, policyDirectory, next) {
             data.names.push(match[1]);
 
             // Add this policy function to the data object
-            var policy = require(path.join(policyDirectory, filename));
-            if (policy.pre === undefined || policy.pre) {
+            var policy = require(path.join(options.policyDirectory, filename));
+            if (policy.applyPoint === undefined || _applyPoints.indexOf(policy.applyPoint) !== -1) {
+                var applyPoint = policy.applyPoint || options.applyPoint;
+
                 server.log(['info'], 'Adding a new PRE policy called ' + match[1]);
-                data.pre[match[1]] = policy;
+                data[applyPoint][match[1]] = policy;
 
-                // connect the prehandler if this is the first pre policy
-                if (!data.setPreHandler) {
-                    server.ext('onPreHandler', preHandler);
-                    data.setPreHandler = true;
-                }
-            }
-            if (policy.post) {
-                server.log(['info'], 'Adding a new POST policy called ' + match[1]);
-                data.post[match[1]] = policy;
-
-                // connect the posthandler if this is the first post policy
-                if (!data.setPostHandler) {
-                    server.ext('onPostHandler', postHandler);
-                    data.setPostHandler = true;
+                // connect the handler if this is the first pre policy
+                if (!data.setHandlers[applyPoint]) {
+                    server.ext(applyPoint, handlers[applyPoint]);
+                    data.setHandlers[applyPoint] = true;
                 }
             }
```

Comment:
Can we add an `else` here to throw an error if you ask for an event apply point that doesn't exist in our array?